airflow/providers/slack/transfers/sql_to_slack.py

```diff
@@ -0,0 +1,162 @@
+    tabulate(tablefmt="pretty", headers="keys") }} to send the results to Slack as an ascii rendered table.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:SqlToSlackOperator`
+
+    :param sql: The SQL statement to execute on Snowflake (templated)
+    :param slack_message: The templated Slack message to send with the data returned from Snowflake.
+        You can use the default JINJA variable {{ results_df }} to access the pandas dataframe containing the
+        SQL results
+    :param sql_conn_id: Reference to
+        :ref:`Snowflake connection id<howto/connection:snowflake>`
+    :param sql_hook_params: Extra config params to be passed to the underlying hook.
+           Should match the desired hook constructor params.
+    :param slack_conn_id: The connection id for Slack.
+    :param slack_webhook_token: The token to use to authenticate to Slack. If this is not provided, the
+        'slack_conn_id' attribute needs to be specified in the 'password' field.
+    :param slack_channel: The channel to send message. Override default from Slack connection.
+    :param results_df_name: The name of the JINJA template's dataframe variable, default is 'results_df'
+    :param parameters: The parameters to pass to the SQL query
+    """
+
+    template_fields: Sequence[str] = ('sql', 'slack_message')
+    template_ext: Sequence[str] = ('.sql', '.jinja', '.j2')
+    template_fields_renderers = {"sql": "sql", "slack_message": "jinja"}
+    times_rendered = 0
+
+    def __init__(
+        self,
+        *,
+        sql: str,
+        sql_conn_id: str,
+        sql_hook_params: Optional[dict] = None,
+        slack_conn_id: str = 'slack_default',
+        slack_webhook_token: Optional[str] = None,
+        slack_channel: Optional[str] = None,
+        slack_message: str,
+        results_df_name: str = 'results_df',
+        parameters: Optional[Union[Iterable, Mapping]] = None,
+        **kwargs,
+    ) -> None:
+
+        super().__init__(**kwargs)
+
+        self.sql_conn_id = sql_conn_id
+        self.sql_hook_params = sql_hook_params
+        self.sql = sql
+        self.parameters = parameters
+        self.slack_conn_id = slack_conn_id
+        self.slack_webhook_token = slack_webhook_token
+        self.slack_channel = slack_channel
+        self.slack_message = slack_message
+        self.results_df_name = results_df_name
+        self.kwargs = kwargs
+
+        if not self.slack_conn_id and not self.slack_webhook_token:
+            raise AirflowException(
+                "SqlToSlackOperator requires either a `slack_conn_id` or a `slack_webhook_token` argument"
+            )
+
+    def _get_hook(self) -> DbApiHook:
+        self.log.debug("Get connection for %s", self.sql_conn_id)
+        conn = BaseHook.get_connection(self.sql_conn_id)
+        hook = conn.get_hook(hook_params=self.sql_hook_params)
```

Comment:
@potiuk need your advise on this one.

The `hook_params` was added in https://github.com/apache/airflow/pull/18718 /  https://github.com/apache/airflow/pull/19849 which were released on `2.3.0` The Slack provider min version is 2.2.0 so I think this code line should have cause failures?

2.2.0:
https://github.com/apache/airflow/blob/2.2.0/airflow/models/connection.py#L292

2.3.0:
https://github.com/apache/airflow/blob/2.3.0/airflow/models/connection.py#L321

 I wonder how it make sense that the CI is green?

@alexkruc in any case we don't need the `hook_params` here.. the goal is just to check if the hook has the needed function, the user params are not needed for that.

Reply:
@alexkruc I think we have two options here.
1. backport the `get_hook` function into the operator code (temporarily, we will remove it once the provider will be Airflow>=2.3 compatible)
2. Leave the `PrestoToSlackOperator` `SnowflakeToSlackOperator` code as is with just a deprecation warning, in that solution we should also check Airflow version when using the operator as it works only with 2.3 (you can check `from airflow.version import version` )

I think option 1 is prefered as otherwise it means only users of Airflow>=2.3 will be able to use this new operator.

Reply:
Yep. 1) is preferred.

Reply:
A fix to pre-commit to make sure we don't have this case is in https://github.com/apache/airflow/pull/24706
Thanks @potiuk

Reply:
@potiuk is there a way to make the new pre commit ignore a specific row?
The pre-commit fails the PR but @alexkruc checks the user Airflow version and have different code to handle base on version.
We can use only the backported version if there is no other option

Reply:
Sure - we can modify the pre-commit and ignore lines that for example contain a comment `# ignore 2.2 comp check`. It's a python code that we can modify in this PR. leaving it as an exercise for the reader :D

Reply:
Reader as in myself? ;)
I'll take a look, will love to get familiar a bit with the pre-commit flow :)
I'll update here if I'll get stuck or something.

I just thought that checking for versions and matching will be easier to read for future operator maintainers, and also it's easier to deprecate later (just remove the if block), I think that using the generic methods (as in `conn.get_hook(hook_params=....)`) as much as possible is the better way here :)

Reply:
@potiuk @eladkal added the ignore lines for compatibility check fix, you can find it in this commit:
https://github.com/apache/airflow/pull/24663/commits/f3a29fb07754e6edaea94ff3e7f81817933958a1
It should run as part of this PR and we can check that it works e2e :) (although it worked locally)

Reply:
Cool! That's what I meant. One small NIT: since you made a "generic" exclusion - let's make it `# ignore airflow compat check`, We have some 2.2, some 2.3 and some 2.4 checks already foreseen to be there for quite some time.